instrumentation/log/filters.go

```diff
@@ -0,0 +1,92 @@
+package log
+
+import "regexp"
+
+type Filter interface {
+	Allows(level string, message string, fields []*Field) bool
+}
+
+func ExcludeField(field *Field) Filter {
+	return &excludeField{field: field}
+}
+
+func IncludeFieldWithKey(key string) Filter {
+	return &includeFieldWithKey{key: key}
+}
+
+func OnlyErrors() Filter {
+	return &levelMatch{level: "error"}
```

Comment:
It shouldn't discard any line that has field `error` populated through `log.Error`.

Reply:
does the log level automatically become "error" if there's an error log? actually, if we use this logic, maybe we can drop the info/error distinction and use the error field heuristic? wdyt?

Reply:
The level does not change because sometimes we want to log the error even in an `info` log because the error is generally not significant. But if we are logging only errors we should still display it.